### PR TITLE
Play Monopoly

### PR DESCRIPTION
**Pull request by dagarcia7**

### CODE REVIEW

File: catanatron/game.py [hunk 1]
```diff
@@ -295,6 +298,22 @@ def execute(self, action, initial_build_phase=False):
...
             player_to_act.development_deck.draw(1, DevelopmentCard.YEAR_OF_PLENTY)
             self.resource_deck -= cards_selected
+        elif action.action_type == ActionType.PLAY_MONOPOLY:
+            player_to_act = action.player
+            card_type_to_steal = action.value
+            cards_stolen = ResourceDeck()
+            if not player_to_act.has_monopoly_card():
+                raise ValueError("Player doesn't have monopoly card")
+            for player in self.players:
+                if not player_to_act.color == player.color:
```
Comment (dagarcia7) [inline catanatron/game.py:308]: thought this was easier than looping with indexes and checking against the `current_player_index` 🤷

File: catanatron/models/player.py [hunk 2]
```diff
@@ -42,6 +42,9 @@ def has_knight_card(self):
     def has_year_of_plenty_card(self):
         return self.development_deck.count(DevelopmentCard.YEAR_OF_PLENTY) > 0
 
+    def has_monopoly_card(self):
```
Comment (dagarcia7) [inline catanatron/models/player.py:45]: I think eventually we can make a more generic (has_dev_card(card_type)) method to replace these

File: tests/test_game.py [hunk 3]
```diff
@@ -192,6 +194,41 @@ def test_play_year_of_plenty_no_year_of_plenty_card():
...
+    players = [SimplePlayer(Color.RED), SimplePlayer(Color.BLUE)]
+    game = Game(players)
+
+    action_to_execute = Action(players[0], ActionType.PLAY_MONOPOLY, Resource.ORE)
+
+    with pytest.raises(ValueError):  # no monopoly
+        game.execute(action_to_execute)
+
+
+def test_play_monopoly_player_steals_cards():
```
Comment (dagarcia7) [inline tests/test_game.py:207]: For this test, I wanted to show that `player_to_act` stole the ORE and left the other resources intact (in this case just WHEAT). I was too lazy to add some other resources in there, but can make this test stronger if necessary

File: tests/models/test_actions.py [hunk 4]
```diff
@@ -36,6 +37,11 @@ def test_year_of_plenty_possible_actions_not_enough_cards():
     assert len(year_of_plenty_possible_actions(player, bank_resource_deck)) == 1
 
 
+def test_monopoly_possible_actions():
+    player = SimplePlayer(Color.RED)
+    assert len(monopoly_possible_actions(player)) == len(Resource)
```
Comment (bcollazo) [inline tests/models/test_actions.py:42]: :fire:

File: tests/test_game.py [hunk 5]
```diff
@@ -192,6 +194,41 @@ def test_play_year_of_plenty_no_year_of_plenty_card():
...
+    game = Game(players)
+    action_to_execute = Action(player_to_act, ActionType.PLAY_MONOPOLY, Resource.ORE)
+
+    game.execute(action_to_execute)
+
+    assert player_to_act.resource_deck.count(Resource.ORE) == 5
+    assert player_to_steal_from_1.resource_deck.count(Resource.ORE) == 0
+    assert player_to_steal_from_1.resource_deck.count(Resource.WHEAT) == 1
+    assert player_to_steal_from_2.resource_deck.count(Resource.ORE) == 0
+    assert player_to_steal_from_2.resource_deck.count(Resource.WHEAT) == 1
```
Comment (bcollazo) [inline tests/test_game.py:229]: :fire: :fire:

File: tests/test_game.py [hunk 6]
```diff
@@ -192,6 +194,41 @@ def test_play_year_of_plenty_no_year_of_plenty_card():
...
+    player_to_act.development_deck.replenish(1, DevelopmentCard.MONOPOLY)
+
+    player_to_steal_from_1.resource_deck.replenish(3, Resource.ORE)
+    player_to_steal_from_1.resource_deck.replenish(1, Resource.WHEAT)
+    player_to_steal_from_2.resource_deck.replenish(2, Resource.ORE)
+    player_to_steal_from_2.resource_deck.replenish(1, Resource.WHEAT)
+
+    players = [player_to_act, player_to_steal_from_1, player_to_steal_from_2]
+    game = Game(players)
+    action_to_execute = Action(player_to_act, ActionType.PLAY_MONOPOLY, Resource.ORE)
```
Comment (bcollazo) [inline tests/test_game.py:221]: Mono Ore. RIP :broken_heart:  :joy:

File: catanatron/game.py (hunk 1 above)
Comment (bcollazo) [reply catanatron/game.py:308]: Looks good to me.

**Review by bcollazo — APPROVED** [review]

**Pull request merged**